### PR TITLE
Refactor 'utils'

### PR DESCRIPTION
**Pull request by Cadair**
This refactors many a util.

It also adds them all to the doc build.

### CODE REVIEW

File: ndcube/tests/test_cube_utils.py [hunk 1]
```diff
@@ -1,69 +1,107 @@
...
-             ['WAVE', 'HPLT-TAN', 'UTC', 'HPLN-TAN'],
-             ['HPLT-TAN', 'TIME', 'HPLN-TAN'],
-             ['HPLT-TAN', 'DEC--TAN', 'WAVE'],
-             [],
-             ['UTC', 'TIME', 'WAVE', 'HPLT-TAN']]
+    lists = [['TIME', 'WAVE', 'HPLT-TAN',
+              'HPLN-TAN'], ['WAVE', 'HPLT-TAN', 'UTC',
+                            'HPLN-TAN'], ['HPLT-TAN', 'TIME', 'HPLN-TAN'],
+             ['HPLT-TAN', 'DEC--TAN',
+              'WAVE'], [], ['UTC', 'TIME', 'WAVE', 'HPLT-TAN']]
```
Comment (DanRyanIrish) [inline ndcube/tests/test_cube_utils.py]: Why is the nice readable layout changed here?

**Comment by DanRyanIrish** [comment]
Huge effort!  Well done!

File: ndcube/tests/test_cube_utils.py (hunk 1 above)
Comment (Cadair) [reply ndcube/tests/test_cube_utils.py]: I auto-layouted it, but this is not nice!

Comment (DanRyanIrish) [reply ndcube/tests/test_cube_utils.py]: What's auto-layout??

Comment (DanRyanIrish) [reply ndcube/tests/test_cube_utils.py]: Is that the same reason the parameterize (input, expected) decorator has exploded?

Comment (Cadair) [reply ndcube/tests/test_cube_utils.py]: https://github.com/google/yapf

yeah, I will give it a tidy

Comment (DanRyanIrish) [reply ndcube/tests/test_cube_utils.py]: I’ve merged this PR and started work on reorganising the testing modules and increasing coverage. Let me open a new PR with my progress so far and you can tidy it there.

Comment (DanRyanIrish) [reply ndcube/tests/test_cube_utils.py]: See #49

**Pull request merged**